src/core/server/deprecations/README.mdx

```diff
@@ -0,0 +1,182 @@
+---
+id: kibUsageCollectionPlugin
+slug: /kibana-dev-docs/services/deprecations-service
+title: Core Deprecations service
+summary: The Deprecations service helps surface deprecated configs and features for plugins to our users
+date: 2021-05-03
+tags: ['kibana','dev', 'contributor', 'api docs']
+---
+
+# Core Deprecations service
+This guide is written in the format of a FAQ to help you get started using the deprecations service.
+For more details on the service contract we've documented all the service apis and properties under the deprecation service docs generated by in core.
+
+## What is the deprecations service?
+The deprecations service provides a way for the Kibana platform to communicate deprecated features and configs with its users. These deprecations are only communicated if the deployment is using these features, allowing for a user tailored experience for upgrading the stack version.
+
+## Where are deprecations displayed to users?
+The Upgrade Assistant (UA) in kibana is consuming the core deprecations service in order to surface plugin deprecations. Each deprecation should have manual steps that we display in the UI to fix the deprecation an API to automatically resolve it.
+
+To check your plugin deprecations go to the UA interface and click on `view deprecations` under the kibana section.
+
+## How do I use this service for deprecated plugin configurations?
+The deprecations service automatically hooks deprecated configs with the deprecations service. 
+
+All the config deprecation functions (`unused`, `unusedFromRoot`, `rename`, `renameFromRoot`) accept an optional parameter to customize the deprecation details.
+
+### Example
+```ts
+export const config: PluginConfigDescriptor<ConfigType> = {
+  schema: configSchema,
+  deprecations: ({ renameFromRoot }) => [
+    renameFromRoot('ui_metric.debug', 'usageCollection.uiCounters.debug', {
+      documentationUrl: 'elastic.co/some-url',
+    }),
+  ],
+};
+```
+
+If the above deprecated kibana config `ui_metric.debug` is being used by the cluster.
+The service will show the following details about the deprecation:
+
+```ts
+{
+  deprecationsInfo:[{
+    level: 'critical',
+    message: `"ui_metric.debug" is deprecated and has been replaced by "usageCollection.uiCounters.debug"`,
+    documentationUrl: 'elastic.co/some-url',
+    correctiveActions:{
+      manualSteps: [
+        `Replace "ui_metric.debug" with "usageCollection.uiCounters.debug" in the Kibana config file, CLI flag, or environment variable (in Docker only).`,
```

Comment:
Should these messages be wrapped in an i18n function for translations?

Reply:
We've had a discussion about this and i'm going to update the docs/faq to include i18n: https://github.com/elastic/kibana/issues/99072